kclvm/tools/src/LSP/src/completion.rs

```diff
@@ -142,34 +107,119 @@ pub(crate) fn completion(
     }
 }
 
-/// Abstraction of CompletionItem in KCL
-#[derive(Debug, Clone, PartialEq, Hash, Eq, Default)]
-pub(crate) struct KCLCompletionItem {
-    pub label: String,
-    pub detail: Option<String>,
-    pub documentation: Option<String>,
-    pub kind: Option<KCLCompletionItemKind>,
-}
-
 fn completion_dot(
     program: &Program,
     pos: &KCLPos,
     prog_scope: &ProgramScope,
+    gs: &GlobalState,
 ) -> Option<lsp_types::CompletionResponse> {
-    // Get the position of trigger_character '.'
-    let pos = &KCLPos {
+    let mut items: IndexSet<KCLCompletionItem> = IndexSet::new();
+
+    let pos = KCLPos {
         filename: pos.filename.clone(),
         line: pos.line,
         column: pos.column.map(|c| c - 1),
     };
 
-    match program.pos_to_stmt(pos) {
-        Some(node) => match node.node {
-            Stmt::Import(stmt) => completion_for_import(&stmt, pos, prog_scope, program),
-            _ => Some(into_completion_items(&get_dot_completion(node, pos, prog_scope)).into()),
-        },
-        None => None,
+    if let Some(stmt) = program.pos_to_stmt(&pos) {
+        match stmt.node {
+            Stmt::Import(stmt) => return completion_import(&stmt, &pos, prog_scope, program),
+            _ => {
+                let (expr, parent) = inner_most_expr_in_stmt(&stmt.node, &pos, None);
```

Comment:
We need to leave a todo here to indicate that the string has not been processed using the new semantic model and will be completed at the cursor inside the string literal instead of at the end.